### PR TITLE
System.Half support

### PR DESCRIPTION
**Pull request by sungam3r**
https://devblogs.microsoft.com/dotnet/introducing-the-half-type/

@SimonCropp I found that I can't write simple assertion test for comparing `Half` types in GraphQL.NET.

### CODE REVIEW

**Review by SimonCropp — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -9,23 +9,23 @@
 Shouldly is an assertion framework which focuses on giving great error messages when the assertion fails while being simple and terse.
 
 This is the old *Assert* way: 
-```cs
+```csharp
```
Comment (Shane32) [inline README.md:12]: There are no reason for this change.  `cs` doesn't work in the GraphQL.NET project for the documentation, but this is a limitation of the GraphQL.NET documentation builder; `cs` works fine for GitHub markdown files.  I always use `cs` when possible.

File: src/Shouldly.Tests/ShouldBe/HalfScenarios.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+{
+    [Fact]
+    public void ShouldPass_SameInstance()
+    {
+        Half h = (Half)12345;
+        h.ShouldBe(h);
+    }
+
+    [Fact]
+    public void ShouldPass_DirrerentInstances()
```
Comment (Shane32) [inline src/Shouldly.Tests/ShouldBe/HalfScenarios.cs:15]:
```suggestion
    public void ShouldPass_DifferentInstances()
```

File: README.md (hunk 1 above)
Comment (sungam3r) [reply README.md:12]: https://github.com/madskristensen/MarkdownEditor is the reason

изображение

изображение

Maybe there are others.

Comment (sungam3r) [reply README.md:12]: The same for https://marketplace.visualstudio.com/items?itemName=MadsKristensen.MarkdownEditor2

These two are the most popular extensions for markdown files
изображение

Comment (Shane32) [reply README.md:12]: Those are both the same; just an older and newer version.   https://github.com/madskristensen/MarkdownEditor2022/issues/80

Comment (sungam3r) [reply README.md:12]: Thanks for posting issue.

**Pull request merged**